File: mikkel_agent/boa_element.py
from geniusweb.issuevalue.Bid import Bid
from geniusweb.profileconnection.ProfileInterface import ProfileInterface

from agents.mikkel_agent.helpers.negotiation_progress import NegotiationProgress


class BoaElement:

    _random_search_phase = 0.75

    def __init__(self, progress: NegotiationProgress, profile: ProfileInterface):
        self._progress = progress
        self._profile = profile

    def get_utility(self, bid: Bid) -> float:
        return self._profile.getProfile().getUtility(bid)

    def get_progress(self) -> float:
        return self._progress.get_progress()

    def is_random_search_phase(self) -> bool:
        return self.get_progress() < self._random_search_phase

File: areaparty/src/main/java/geniusweb/exampleparties/areaparty/BoaElement.java
package geniusweb.exampleparties.areaparty;

import geniusweb.issuevalue.Bid;
import geniusweb.profile.utilityspace.LinearAdditiveUtilitySpace;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.experimental.SuperBuilder;

@SuperBuilder
@RequiredArgsConstructor
public class BoaElement {

    private static final double RANDOM_SEARCH_PHASE = 0.75;
    private static final double CONCESSION_PHASE = 0.9;

    private final NegotiationProgress progress;

    @Getter
    private final LinearAdditiveUtilitySpace profile;

    public double getUtility(Bid bid) {
        return this.profile.getUtility(bid).doubleValue();
    }

    public double getProgress() {
        return this.progress.getProgress();
    }

    public boolean isRandomSearchPhase() {
        return this.getProgress() < RANDOM_SEARCH_PHASE;
    }

    public boolean isConcessionPhase() {
        return this.getProgress() > CONCESSION_PHASE;
    }
}
